File: java/src/adapter/ClassAdapterExample/VenderFactory.java
package adapter.ClassAdapterExample;

public class VenderFactory {
	private static VenderFactory instance;
	private VenderFactory() {}	
	public static VenderFactory getInstance() {
		if (instance == null) 
			instance = new VenderFactory();
		return instance;
	}
	public Vender createVender() {
		return new VenderClassAdapter();
	//	return new Vender();
	//	return new VenderAdapter( new Vender2());
	}

}

File: src/adapter/ClassAdapterExample/VenderFactory.kt
package adapter.ClassAdapterExample

class VenderFactory private constructor() {
    fun createVender(): Vender {
        return VenderClassAdapter()
        //	return new Vender();
        //	return new VenderAdapter( new Vender2());
    }

    companion object {
        var instance: VenderFactory? = null
            get() {
                if (field == null) field = VenderFactory()
                return field
            }
            private set
    }
}
